src/bgw/job_stat_history.c

```diff
@@ -53,19 +113,26 @@ ts_bgw_job_stat_history_insert(BgwJob *job, BgwJobStatHistoryContext *context)
 								 values,
 								 ts_timer_get_current_timestamp(),
 								 false);
-		ts_datum_set_jsonb(Anum_bgw_job_stat_history_error_data, values, context->edata);
+		ts_datum_set_jsonb(Anum_bgw_job_stat_history_data,
+						   values,
+						   ts_bgw_job_stat_history_build_data_info(context->job, context->edata));
 	}
 	else
-		ts_datum_set_jsonb(Anum_bgw_job_stat_history_error_data, values, NULL);
+	{
+		ts_datum_set_jsonb(Anum_bgw_job_stat_history_data,
+						   values,
+						   ts_bgw_job_stat_history_build_data_info(context->job, NULL));
```

Comment:
I am wondering if this works as expected. According to the condition of the branch `context` should be NULL at this point.

Reply:
I agree this looks odd and at least would need a test.

Reply:
Make sense... simplified the code and use only the `context`. The `ts_bgw_job_stat_history_build_data_info` knows how to deal with NULL information